backend/src/polydash/polygon/top_peers.py

```diff
@@ -0,0 +1,60 @@
+from fastapi import APIRouter
+from pony.orm import desc, db_session
+
+from polydash.miners_ratings.model import MinerRisk
+from polydash.polygon.deanon.model import DeanonNodeByBlock, DeanonNodeByTx, PeerToIP
+
+TOP_NODES_LIST_SIZE = 10
+
+top_peers_router = router = APIRouter(
+    prefix="/top-peers",
+    tags=["W3Router"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/")
+@db_session
```

Comment:
You are not allowed to decorate `async` functions with `@db_session`, last paragraph on this page: https://docs.ponyorm.org/integration_with_fastapi.html. On my PC it can segfault or return an empty array even when some data is provided.

So, you need to use `with db_session` in the body of the function.

Reply:
Good find, missed that :+1: